### PR TITLE
Add module crm_lead_supplier.

### PR DESCRIPTION
**Pull request by yajo**
Relaunching last part of Antiun/antiun-odoo-addons#11.

Add customer/supplier checkbox fields to leads.

It will fully work after odoo/odoo#8899 is merged, but I think it can be merged anyway.

cc @rafaelbn.

### CODE REVIEW

**Comment by yajo** [comment]
Now NOT depending on odoo/odoo#8899.

**Comment by rafaelbn** [comment]
Tested in runbot :+1:

**Comment by pedrobaeza** [comment]
Please check Travis

**Comment by yajo** [comment]
I cannot check Travis, it's stuck loading.

File: crm_lead_supplier/README.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/crm/issues>`_. In
+case of trouble, please check there if your issue has already been reported. If
+you spotted it first, help us smashing it by providing a detailed and welcomed
+feedback `here <https://github.com/OCA/ crm/issues/new?body=module:%20
+crm_lead_website%0Aversion:%20
+8.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (osvalr) [inline crm_lead_supplier/README.rst]: the version shouldn't be the same on manifest file?

**Comment by osvalr** [comment]
`class Lead` is in `crm_lead_supplier/models.py` instead `crm_lead_supplier/models/lead.py`

**Comment by dgzurita** [comment]
Tested in runbot :+1:

**Comment by yajo** [comment]
@osvalr That adds redundancy for such a simple module IMHO.

**Comment by yajo** [comment]
Can this be a Travis error? [Seems like `new_api == None`](https://travis-ci.org/OCA/crm/jobs/86186792#L287), but that makes no sense, and works in localhost.

**Comment by yajo** [comment]
Seems like https://github.com/OCA/crm/pull/52#issuecomment-148742486, but this one is rebased already.

**Comment by osvalr** [comment]
Hi @Yajo 
You have to read [File naming](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#file-naming) section of OCA

**Comment by janetrp** [comment]
when creating a lead this error shows

Odoo

Odoo Server Error

Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 537, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 574, in dispatch
    result = self._call_function(*_self.params)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 310, in _call_function
    return checked_call(self.db, *args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/service/model.py", line 118, in wrapper
    return f(dbname, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 307, in checked_call
    return self.endpoint(_a, *_kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 803, in __call__
    return self.method(_args, *_kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 403, in response_wrap
    response = f(_args, *_kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/web/controllers/main.py", line 944, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/web/controllers/main.py", line 936, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 250, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 372, in old_api
    result = method(recs, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/models.py", line 5929, in onchange
    record._onchange_eval(name, field_onchange[name], result)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/models.py", line 5836, in _onchange_eval
    method_res = getattr(self._model, method)(_args)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 372, in old_api
    result = method(recs, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/crm_location/models/crm_lead.py", line 49, in on_change_partner_id
    res = super(CrmLead, self).on_change_partner_id(partner_id)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 248, in wrapper
    return new_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/crm_lead_supplier/models/crm_lead.py", line 26, in on_change_partner_id
    result = super(Lead, self).on_change_partner_id(partner_id)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 248, in wrapper
    return new_api(self, _args, *_kwargs)
TypeError: 'NoneType' object is not callable

When closing the extra info tab is showed and the check box supplier is in the tab
2015-10-22 16_54_36-leads - odoo.

then checked and converting to opportunity same error shows:
Odoo

Odoo Server Error

Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 537, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 574, in dispatch
    result = self._call_function(*_self.params)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 310, in _call_function
    return checked_call(self.db, *args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/service/model.py", line 118, in wrapper
    return f(dbname, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 307, in checked_call
    return self.endpoint(_a, *_kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 803, in __call__
    return self.method(_args, *_kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/http.py", line 403, in response_wrap
    response = f(_args, *_kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/web/controllers/main.py", line 948, in call_button
    action = self._call_kw(model, method, args, {})
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/web/controllers/main.py", line 936, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 250, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/crm/wizard/crm_lead_to_opportunity.py", line 161, in action_apply
    self._convert_opportunity(cr, uid, ids, vals, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 250, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/crm/wizard/crm_lead_to_opportunity.py", line 119, in _convert_opportunity
    partner_id = self._create_partner(cr, uid, lead_id.id, data.action, partner_id or lead_id.partner_id.id, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 250, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/crm/wizard/crm_lead_to_opportunity.py", line 181, in _create_partner
    res = lead.handle_partner_assignation(cr, uid, [lead_id], action, partner_id, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 250, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/crm/crm_lead.py", line 798, in handle_partner_assignation
    partner_id = self._create_lead_partner(cr, uid, lead, context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 250, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/crm/crm_lead.py", line 766, in _create_lead_partner
    partner_id = self._lead_create_contact(cr, uid, lead, lead.partner_name, True, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 250, in wrapper
    return old_api(self, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 345, in old_api
    result = method(recs, _args, *_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/addons/crm_lead_supplier/models/crm_lead.py", line 21, in _lead_create_contact
    ._lead_create_contact(lead, name, is_company, parent_id))
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3117988-55-e220d3/openerp/api.py", line 248, in wrapper
    return new_api(self, _args, *_kwargs)
TypeError: 'NoneType' object is not callable

so, can't test

**Comment by yajo** [comment]
That error is being worked on in https://github.com/OCA/crm/pull/52, it's being a bit tricky. Once fixed there, it will be fixed everywhere.

**Comment by pedrobaeza** [comment]
:+1:

**Pull request merged**